nodes/matrix/matrix_in_mk3.py

```diff
@@ -82,45 +61,17 @@ def update_mode(self, context):
 
         updateNode(self, context)
 
-    def update_angle_units(self, context):
+    def update_angles(self, context, auc):
         ''' Update all the angles to preserve their values in the new units '''
-
-        auc = angle_unit_conversion[self.last_angle_units][self.angle_units]
-
-        self.last_angle_units = self.angle_units  # keep track of the last units
-
-        self.syncing = True  # deactivate updates
         self.angle = self.angle * auc
         self.angle_x = self.angle_x * auc
         self.angle_y = self.angle_y * auc
         self.angle_z = self.angle_z * auc
-        self.syncing = False  # reactivate updates
-
-        updateNode(self, context)
-
-    def update_angle(self, context):
-        ''' Wrapper to suppress angle updates when units are changed '''
-        if self.syncing:
-            return
-
-        updateNode(self, context)
 
     mode: EnumProperty(
```

Comment:
would you not stick this inside the helper class, but use a more descriptive property name other than `mode`..maybe `axis_mode` ... hopefully it won't be too painful